tools/tu_collector/tu_collector/tu_collector.py

```diff
@@ -281,6 +281,10 @@ def zip_tu_files(zip_file, compilation_database, write_mode='w'):
 
     add_sources_to_zip(zip_file, tu_files)
 
+    with zipfile.ZipFile(zip_file, 'a') as archive:
```

Comment:
Also we open the zipfile multiple times if some errors occurred:
https://github.com/Ericsson/codechecker/blob/e09c190e54cb6e70eb1e86f1d538adbc27cb69f8/tools/tu_collector/tu_collector/tu_collector.py#L278-L280

We can open it only once:
```py
with zipfile.ZipFile(zip_file, write_mode) as archive:
    archive.writestr('compilation_database.json',
                     json.dumps(compilation_database, indent=2))

    if error_messages:
        archive.writestr(no_sources, error_messages)
```